Bottom: ba1e6b42da47b5da8575ac2b2c88b824e7c688dc
Top:    cd25cd858812228fa98f54563a94aac664282722
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-10-03 23:22:16 +0100

Refresh of play-with-glsl-shader-for-rend

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 752cbe5..b07b256 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -338,10 +338,14 @@ static void
 ghid_draw_bg_image (void)
 {
   static GLuint texture_handle = 0;
+  GLuint current_program;
 
   if (!ghidgui->bg_pixbuf)
     return;
 
+  glGetIntegerv (GL_CURRENT_PROGRAM, (GLint*)&current_program);
+  glUseProgram (0);
+
   if (texture_handle == 0)
     {
       int width =             gdk_pixbuf_get_width (ghidgui->bg_pixbuf);
@@ -389,6 +393,8 @@ ghid_draw_bg_image (void)
   glEnd ();
 
   glDisable (GL_TEXTURE_2D);
+
+  glUseProgram (current_program);
 }
 
 void
